decidim-bulletin_board-js/src/voter/voter.js

```diff
@@ -14,21 +15,42 @@ export class Voter {
    * @param {Object} params - An object that contains the initialization params.
    *  - {String} id - The voter identifier.
    */
-  constructor({ id, bulletinBoardClient, electionContext, options }) {
+  constructor({ id, electionContext, bulletinBoardClient, options }) {
     this.id = id;
-    this.wrapper = new VoterWrapper({ voterId: id });
-    this.bulletinBoardClient = bulletinBoardClient;
     this.electionContext = electionContext;
+    this.bulletinBoardClient = bulletinBoardClient;
+    this.wrapper = new VoterWrapper({ voterId: id });
+    this.parser = new JWTParser();
     this.options = options || { bulletinBoardWaitTime: WAIT_TIME_MS };
   }
 
+  /**
+   * Performs some operations to setup the voter.
+   *
+   * Retrieves the key ceremony messages needed to cast a vote in the given election.
+   *
+   * @returns {Promise<undefined>}
+   */
+  setup() {
+    return this.bulletinBoardClient
+      .getElectionLogEntries({
+        electionUniqueId: this.electionContext.id,
+      })
+      .then(async (logEntries) => {
+        for (const logEntry of logEntries) {
+          const message = await this.parser.parse(logEntry.signedData);
```

Comment:
Do we need to ensure that the `logEntry.signedData` is present?

Reply:
We could do it, but now the messages without `signedData` are the tally ones, so there shouldn't be a problem at the moment.